src/gui/torrentcontentmodel.cpp

```diff
@@ -88,6 +98,43 @@ namespace
             return QIcon(iconPixmap);
         }
     };
+#elif defined(Q_OS_MAC)
+    // There is a similar bug on macOS, to be reported to Qt
+    // https://github.com/qbittorrent/qBittorrent/pull/6156#issuecomment-316302615
+    class MacFileIconProvider: public UnifiedFileIconProvider
+    {
+    public:
+        using QFileIconProvider::icon;
+        QIcon icon(const QFileInfo &info) const override
+        {
+            QString suffix = info.suffix();
```

Comment:
`const auto` is OK, if you like. `const` is to point out that you are not going to change it below. Helps to read. You might object that the function is so short that it is not necessary to simplify reading, but we use `const` with this intent in other places, and when one encounters (in this project) a variable without a `const`, it a fat hint that ti is going to be changed below, that slows down code understanding.